libmscore/style.cpp

```diff
@@ -2474,18 +2534,221 @@ void MStyle::save(XmlWriter& xml, bool optimize)
+      top    = value(Sid::marginOddTop).toInt()    / factor;
+      bottom = value(Sid::marginOddBottom).toInt() / factor;
+cout << "  toPageLayout - Odd Left Origin: " << left << endl;
+
+      QMarginsF oddMarg = QMarginsF(left, top, right, bottom);
+      // For two-sided pages: even L/R margins are a mirror-image of odd margins
+      // For one-sided pages: even L/R margins are identical to odd margins
+      if (isTwo) {
+            double tmp = left;
+            left  = right;
+            right = tmp;
+            }
+      top    = value(Sid::marginEvenTop).toInt()    / factor;
+      bottom = value(Sid::marginEvenBottom).toInt() / factor;
+      QMarginsF evenMarg = QMarginsF(left, top, right, bottom);
+
+      QPageLayout::Orientation orient;
+      if (_isMMInch) {
+            if (psid == QPageSize::Custom)
+                  orient = (h >= w ? QPageLayout::Portrait : QPageLayout::Landscape);
+            else {
+                  QSizeF size = _pageSize.definitionSize();
+                  if ((w > h) == (size.width() > size.height())) 
+                        orient = QPageLayout::Portrait;
+                  else
+                        orient = QPageLayout::Landscape;
+                  }
+            set(Sid::pageOrientation, pageOrient[int(orient)]);
+            _isMMInch = false; // no longer needed
+            }
+      else
+            orient = (value(Sid::pageOrientation).toString() == pageOrient[1])
+                     ? QPageLayout::Landscape
+                     : QPageLayout::Portrait; // defaults to portrait
+
+      _pageOdd = MPageLayout();
+      _pageOdd.setPageSize(_pageSize);
+      _pageOdd.setUnits(QPageLayout::Unit(idxUnit));
+      _pageOdd.setMargins(oddMarg);
+cout << "  toPageLayout - Odd Left Source: " <<  oddMarg.left() << endl;
+cout << "  toPageLayout - Odd Left Margin: " << _pageOdd.margins(QPageLayout::Inch).left() << endl;
+      _pageOdd.setOrientation(orient);
+
+      _pageEven = MPageLayout(_pageOdd);
+      _pageEven.setMargins(evenMarg);
+      }
+
+//---------------------------------------------------------
+//   isMetric
+//---------------------------------------------------------
+
+bool MStyle::isMetric(QPageLayout::Unit unit)
+      {
+      switch (unit) {
+            case QPageLayout::Millimeter:
+            case QPageLayout::Didot:
+            case QPageLayout::Cicero:
+                  return true;
+            default:
+                  return false;
+            }
+      }
+
 //---------------------------------------------------------
-//   reset
+//   fuzzyCompare
 //---------------------------------------------------------
 
-void MStyle::reset(Score* score)
+bool MStyle::fuzzyCompare(Sid sid, double pageValue)
       {
-      for (const StyleType& st : styleTypes)
-            score->undo(new ChangeStyleVal(score, st.styleIdx(), MScore::defaultStyle().value(st.styleIdx())));
+      return(abs(value(sid).toDouble() - pageValue) >= 0.01);
```

Comment:
Please replace `0.01` with `epsilon` variable which should be a parameter of the method with default value `0.01`.

Reply:
In the next push, later tonight.